dapp/src/utils/evm/chains.ts

```diff
@@ -8,19 +8,44 @@ import {
   sepolia,
 } from "viem/chains";
 
+/**
+ * Define supported chains
+ */
 export const supportedChains: Chain[] = env.REACT_APP_TESTNET
   ? [sepolia, arbitrumSepolia]
   : [mainnet];
 
-export const isChainSupported = (chainId: number | undefined) => {
-  return (
-    chainId != null && !!supportedChains.find((chain) => chain.id === chainId)
-  );
-};
-
+/**
+ * Mapping of numerical chain ids to Alchemy Network object
+ */
 export const WagmiToAlchemy: Record<number, Network> = {
   [sepolia.id]: Network.ETH_SEPOLIA,
   [mainnet.id]: Network.ETH_MAINNET,
   [arbitrum.id]: Network.ARB_MAINNET,
   [arbitrumSepolia.id]: Network.ARB_SEPOLIA,
 };
+
+/**
+ * Define reserve waiting time that is added to the actual unlock time,
+ * in order to not offer withdraw operation before there is a block surpassing the unlock time
+ */
+export const getChainReserveWaitingTime = (
+  chainId: number | undefined,
+): bigint => {
+  switch (chainId) {
+    case sepolia.id:
+    case mainnet.id:
+    case arbitrum.id:
+    case arbitrumSepolia.id:
+      // 1.5x blocktime (12s)
+      return (12n * 3n) / 2n;
+    default:
+      return 0n;

```

Comment:
is returning `0` actually safe here? The comment for this function makes it sound like `0` is the one value that is guaranteed not to give the ideal behavior on any chain

Reply:
It's not "unsafe" per se. It would just mean that you if you click withdraw too soon, the button will bounce back to default state and you get a console error of `Error: NotUnlockedYet()`. So would result in bad UX that would probably get reported. This constant is in one of few config files that are also described in readme, so I feel like anybody implementing should not forget to set this either way. I wouldn't like to put some arbitrary number here, as that would probably just be confusing.

Reply:
okay, we should at least have a comment in the code explaining this rationale for picking `0n` then